core/src/test/scala/unit/kafka/integration/MetricsDuringTopicCreationDeletionTest.scala

```diff
@@ -59,9 +59,8 @@ class MetricsDuringTopicCreationDeletionTest extends KafkaServerTestHarness with
     for (m <- (testedMetrics)) {
         Metrics.defaultRegistry.allMetrics.asScala
         .filterKeys(k => k.getName.endsWith(m))
-        .headOption match {
-           case Some(e) => Metrics.defaultRegistry.removeMetric(e._1)
-           case None =>
+        .headOption.foreach { e =>
+          Metrics.defaultRegistry.removeMetric(e._1)
         }
```

Comment:
Sorry my bad, you can ignore this one. The same compaction cannot be applied here.